### PR TITLE
Related #13336 - DNS resolution is preventing awx_collection to work with http[s]_proxy

### PR DESCRIPTION
**Pull request by bagasse**
##### SUMMARY

ControllerApi code check for host resolution via getaddrinfo. This prevent any module to work behind http[s]_proxy when hostnames are not resolvable on the host where the code is run. See #13336 for more info.

##### ISSUE TYPE

 - Bug

##### COMPONENT NAME

 - Collection

##### AWX VERSION

```
awx: 0.1.dev32821+g0a6e0ae
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by bagasse** [comment]
The other solution should be to remove this DNS test.

**Comment by fosterseth** [comment]
@bagasse what controller url and port were you providing that fails this getaddrinfo check? if you cannot provide the exact url, just an equivalent form is fine

**Comment by bagasse** [comment]
Hello @fosterseth, 
Thanks to take a look at this. The `controller_host` param that I used only contain the hostname of the controller (eg: aac.xxx.redhat.com), the module convert it in https://aac.xxx.redhat.com.

The thing is, when you use a HTTP(s) proxy, the DNS resolution is made by the proxy (not by the host that make the http request), and so in some case, only the http proxy is able to resolve the DNS name (that the case of my lab build on an internal Red Hat lab infra), so in that case the current test will throw an error for a valid host. In some company with a lot of different security zones and internal http proxy, it can be a pain.

The other solution/implementation should be to remove this dns resolution test, because if it's not resolvable the underlying http library will return the same error anyway.

**Comment by fosterseth** [comment]
@shanemcd I think this is a valid case.

should we just remove the hostname DNS resolution gut check entirely and let the module fail further down below?

File: awx_collection/plugins/module_utils/controller_api.py
Comment (TheRealHaoLiu) [inline awx_collection/plugins/module_utils/controller_api.py:135]:
```
TASK [Create Credential for target cluster in AWX] **************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Unable to resolve controller_host (name 'environ' is not defined): gabe.something.com"}
```

Comment (TheRealHaoLiu) [reply awx_collection/plugins/module_utils/controller_api.py:135]: `from os import access, R_OK, getcwd, environ` need to be added

**Comment by TheRealHaoLiu** [comment]
tested the PR with the following scenarios 

with HTTPS_PROXY set and unresolvable hostname, error does not show
with HTTPS_PROXY not set and unresolvable hostname, error show `"Unable to resolve controller_host ([Errno 8] nodename nor servname provided, or not known): yoloswasdfasdfasfasfasdgabeag.com"`
with HTTPS_PROXy not set and resolvable hostname error not show 

after you fix the import we can merge it

**Comment by TheRealHaoLiu** [comment]
please update your PR with description and version information

**Comment by fosterseth** [comment]
@TheRealHaoLiu correct import statement now, want to do a final review? can probably merge soon

**Review by fosterseth — APPROVED** [review]

**Pull request merged**
